dask/dataframe/core.py

```diff
@@ -3292,6 +3292,14 @@ def __len__(self):
         else:
             return len(s)
 
+    @property
+    def empty(self):
+        raise NotImplementedError(
+            "Checking the dimensions of a Dask DataFrame can be expensive. "
```

Comment:
The only thing I would clarify here is that only checking whether there are any rows is expensive. Checking the number of columns is fast.